include/deal.II/grid/grid_generator.h

```diff
@@ -60,6 +61,179 @@ namespace GridGenerator
+       */
+      double length_b2;
+
+      /**
+       * Factor defining the inclination of the face that block 2 and 3
+       * or block 5 and 6 share.
+       */
+      double incline_factor;
+
+      /**
+       * Factor to receive a finer mesh around the airfoil by increasing it.
+       */
+      double bias_factor;
+
+      /**
+       * Global refinement-level.
+       */
+      int refinements;
+
+      /**
+       * Number of subdivisions along the airfoil in left block.
+       */
+      int n_subdivision_x_0;
+
+      /**
+       * Number of subdivisions along the airfoil in middle block.
+       */
+      int n_subdivision_x_1;
+
+      /**
+       * Number of subdivisions in block right of the airfoil.
+       */
+      int n_subdivision_x_2;
+
+      /**
+       * Number of subdivisions normal to the airfoil contour.
+       */
+      int n_subdivision_y;
+
+      /**
+       * Factor to enhance the approximation of the airfoil geometry that
+       * happens when interpolating provided nonequidistant airfoil points to
+       * equidistant airfoil points. When generating the required vector
+       * consisting the equidistant airfoil points, function
+       * make_points_equidistant interpolates between nonequidistand airfoil
+       * points. Increasing the provided nonequidistant airfoil points leads to
+       * a better approximation of the airfoil geometry. Parameter
+       * "point_factor" therby defines the relation of
+       * provided_nonequidistant_points to required_equidistant_points.
+       */
+      unsigned int point_factor;
+
+      /**
+       * Constructor.
+       */
+      AdditionalData();
+
+      /**
+       * This Function adds the ParameterHandler entries.
+       *
+       * @param prm Parameter handler.
+       */
+      void
+      addparameter(ParameterHandler &prm);
+    };
+
+
+
+    /**
+     * Initialize the given triangulation with a flow field around an airfoil,
+     * i.e. a Mesh of C-Type approximating Joukowski or NACA (4 digit) airfoils.
+     *
+     * To specify the airfoil geometry and the mesh setup the provided
+     * inputparametrs from class AdditionalData is used. Thereby, different
+     * types of Joukowski or NACA airfoils with variable chord length, far field
+     * size and mesh density can be created.
+     *
+     * By default, a triangulation for a NACA 2412 airfoil with chord
+     * lenght one is generated.
+     *
+     * @note This function creates a refined mesh (number of global refinements
+     *       can be specified by the user). No manifold is attached. The
+     *       vertices in the final meshed are moved by this function to the
+     *       right position.
+     *
+     * @note This function is currently only implemented for 2D.
+     *
+     * @param tria The triangulation to be created. It needs to be empty upon
+     *             calling this function.
+     * @param additional_data Configuration of the mesh.
```

Comment:
So a figure created in incscape showing a mesh schematically with the input parametes of class AdditionalData, right? I will add the figure till Friday evening.
Which format of the picture, png? And how to add a picture in the code? By the link where it is saved?

Reply:
Regarding the figure, I suggest we post it on the homepage (since it is 364 kB, so a bit large to include it in the git repo). I have uploaded the file to the homepage repo, the link is:
https://www.dealii.org/images/grids/airfoils_naca_joukowski.png

Given the resolution of the file, I suggest we insert the file with some relative width compared to screen. I'm not an expert on what markup is allowed by doxygen, but looking around in other places suggests we could do something like
```
\htmlonly <style>div.image
   * img[src="https://www.dealii.org/images/grids/airfoils_naca_joukowski.png"]{width:50%;}</style> \endhtmlonly
   * @image html https://www.dealii.org/images/grids/airfoils_naca_joukowski.png
```